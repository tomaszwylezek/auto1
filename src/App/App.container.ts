import { connect } from 'react-redux';
import { getMerchants } from 'redux/merchants/merchants.actions';

import { App } from './App';

import { IAppConnectedActions, IAppConnectedState } from './App.types';

const mapStateToProps = (state): IAppConnectedState => ({});

const mapDispatchToProps: IAppConnectedActions = {
  getMerchants
};

export const AppContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
