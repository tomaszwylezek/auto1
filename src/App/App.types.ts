export interface IAppOwnProps {}

export interface IAppConnectedState {}

export interface IAppConnectedActions {
  getMerchants: () => void;
}

export interface IAppProps extends IAppConnectedState, IAppConnectedActions, IAppOwnProps {}

export interface IAppState {}
