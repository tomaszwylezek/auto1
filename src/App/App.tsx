import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import { Nav } from 'App/Nav/Nav';

// Models
import { IAppProps } from 'App/App.types';

export class App extends Component<IAppProps> {

  componentDidMount(): void {
    this.props.getMerchants();
  }

  render(): JSX.Element {

    return <Router>
        <Nav />
        <main className="container py-4">
          <Switch>
            <Route path="/" children={() => 'main'} />
          </Switch>
        </main>
      </Router>;
  }
}
