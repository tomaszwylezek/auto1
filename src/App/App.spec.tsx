import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { App } from './App';

// Models
import { IAppProps, IAppState } from './App.types';

describe('App component should', (): void => {
  let props: IAppProps;
  let component: ShallowWrapper<IAppProps, IAppState>;

  beforeEach((): void => {
    props = {
      getMerchants: jest.fn()
    };

    component = shallow(<App {...props} />);
  });

  it('render correctly', (): void => {
    expect(component).toMatchSnapshot();
  });

  it('calls getMerchants onMount', (): void => {
    expect(props.getMerchants).toHaveBeenCalled();
  });
});
