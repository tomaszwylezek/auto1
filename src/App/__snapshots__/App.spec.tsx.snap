// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly 1`] = `
<BrowserRouter>
  <Nav />
  <main
    className="container py-4"
  >
    <Switch>
      <Route
        path="/"
      >
        <Component />
      </Route>
    </Switch>
  </main>
</BrowserRouter>
`;
